tests/gocase/unit/introspection/replication_test.go

```diff
@@ -0,0 +1,349 @@
+	util.Populate(t, masterClient, "", 1024, 1)
+	ctx := context.TODO()
+	require.NoError(t, masterClient.Set(ctx, "a", "b", 0).Err())
+	require.NoError(t, masterClient.Do(ctx, "compact").Err())
+	time.Sleep(time.Second)
+
+	slave1 := util.StartServer(t, map[string]string{})
+	defer slave1.Close()
+	slave1Client := slave1.NewClient()
+	defer func() { require.NoError(t, slave1Client.Close()) }()
+	util.Populate(t, slave1Client, "", 1026, 1)
+
+	slave2 := util.StartServer(t, map[string]string{})
+	defer slave2.Close()
+	slave2Client := slave2.NewClient()
+	defer func() { require.NoError(t, slave2Client.Close()) }()
+	util.Populate(t, slave2Client, "", 1026, 1)
+
+	t.Run("two slaves share one checkpoint for full replication", func(t *testing.T) {
+		util.SlaveOf(t, slave1Client, master)
+		util.SlaveOf(t, slave2Client, master)
+
+		require.Eventually(t, func() bool {
+			return master.LogFileMatches(t, ".*Use current existing checkpoint.*")
+		}, 50*time.Second, 100*time.Millisecond)
+		util.WaitForSync(t, slave1Client)
+		util.WaitForSync(t, slave2Client)
+		require.Equal(t, "b", slave1Client.Get(ctx, "a").Val())
+		require.Equal(t, "b", slave2Client.Get(ctx, "a").Val())
+	})
+}
+
+func TestReplicationContinueRunning(t *testing.T) {
+	master := util.StartServer(t, map[string]string{})
+	defer master.Close()
+	masterClient := master.NewClient()
+	defer func() { require.NoError(t, masterClient.Close()) }()
+
+	slave := util.StartServer(t, map[string]string{})
+	defer slave.Close()
+	slaveClient := slave.NewClient()
+	defer func() { require.NoError(t, slaveClient.Close()) }()
+
+	util.SlaveOf(t, slaveClient, master)
+	util.WaitForSync(t, slaveClient)
+
+	t.Run("Master doesn't pause replicating with replicas, #346", func(t *testing.T) {
+		ctx := context.TODO()
+		// In #346, we find a bug, if one command contains more than special
+		// number updates, master won't send replication stream to replicas.
+		masterClient.HSet(ctx, "myhash", 0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9,
+			"a", "a", "b", "b", "c", "c", "d", "d", "e", "e", "f", "f", "g", "g", "h", "h", "i", "i", "j", "j", "k", "k")
+		require.EqualValues(t, 21, masterClient.HLen(ctx, "myhash").Val())
+		util.WaitForOffsetSync(t, masterClient, slaveClient)
+		require.Equal(t, "1", slaveClient.HGet(ctx, "myhash", "1").Val())
+		require.Equal(t, "a", slaveClient.HGet(ctx, "myhash", "a").Val())
+	})
+}
+
+func TestReplicationChangePassword(t *testing.T) {
+	master := util.StartServer(t, map[string]string{})
+	defer master.Close()
+	masterClient := master.NewClient()
+	defer func() { require.NoError(t, masterClient.Close()) }()
+
+	slave := util.StartServer(t, map[string]string{})
+	defer slave.Close()
+	slaveClient := slave.NewClient()
+	defer func() { require.NoError(t, slaveClient.Close()) }()
+
+	util.SlaveOf(t, slaveClient, master)
+	util.WaitForSync(t, slaveClient)
+
+	t.Run("Slave can re-sync with master after password change", func(t *testing.T) {
+		ctx := context.TODO()
```

Comment:
```suggestion
		ctx := context.Background()
```

nit: consistently using `context.Background()`.

Reply:
> Unstable case. Please take a look.
> 
> ```
> --- FAIL: TestReplicationChangePassword (2.74s)
>     --- FAIL: TestReplicationChangePassword/Slave_can_re-sync_with_master_after_password_change (0.10s)
>         replication_test.go:346: 
>             	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/introspection/replication_test.go:346
>             	Error:      	"info replication: # Replication\r\nrole:master\r\nconnected_slaves:1\r\nmaster_repl_offset:0\r\n" does not contain "127.0.0.1"
>             	Test:       	TestReplicationChangePassword/Slave_can_re-sync_with_master_after_password_change
> ```

ok, will inspect the unstable reason.